src/dbnode/storage/index/block.go

```diff
@@ -794,7 +794,7 @@ func (b *block) addAggregateResults(
 	batch []AggregateResultsEntry,
 ) ([]AggregateResultsEntry, int, int, error) {
 	// update recently queried docs to monitor memory.
-	if err := b.queryStats.Update(len(batch)); err != nil {
+	if err := b.queryStats.UpdateDocs(len(batch)); err != nil {
```

Comment:
nit: Might be nice to move this check up earlier in the aggregation path as well, but up to you

Reply:
Good call